src/AspNetCoreModuleV2/CommonLib/debugutil.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "debugutil.h"
+#include "dbgutil.h"
+
+inline HANDLE g_hStandardOutput;
+
+VOID
+DebugInitialize()
+{
+    g_hStandardOutput = GetStdHandle(STD_OUTPUT_HANDLE);
+
+    HKEY hKey;
+
+    if (RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+            L"SOFTWARE\\Microsoft\\IIS Extensions\\IIS AspNetCore Module\\Parameters",
+            0,
+            KEY_READ,
+            &hKey) == NO_ERROR)
+    {
+        DWORD dwType;
+        DWORD dwData;
+        DWORD cbData;
+
+        cbData = sizeof(dwData);
+        if ((RegQueryValueEx(hKey,
+            L"DebugFlags",
+            NULL,
+            &dwType,
+            (LPBYTE)&dwData,
+            &cbData) == NO_ERROR) &&
+            (dwType == REG_DWORD))
+        {
+            DEBUG_FLAGS_VAR = dwData;
+        }
+
+        RegCloseKey(hKey);
+    }
+
+    // We expect single digit value and a null char
+    const size_t environmentVariableValueSize = 2;
+    std::wstring environmentVariableValue(environmentVariableValueSize, '\0');
+
+    if (GetEnvironmentVariable(L"ASPNETCORE_MODULE_DEBUG", environmentVariableValue.data(), environmentVariableValueSize) == environmentVariableValueSize - 1)
```

Comment:
nit formatting.

Reply:
Also, are .data() and &env[0] the same?

Reply:
Yes, everyone was using `[0]` because `data` was const until recently